vllm/distributed/parallel_state.py

```diff
@@ -673,6 +677,31 @@ def model_parallel_is_initialized():
     return (_TP is not None and _PP is not None)
 
 
+OVERRIDE_TP_STATE = False
+
+
+@contextlib.contextmanager
+def patch_tensor_parallel_group(world_group, tp_group):
```

Comment:
Will this global variable patching potentially create problem? For example, is it possible that other workers will use this context unknowingly?

Reply:
In the current design of speculative decoding, draft and target workers execute _sequentially_.
So there's no chance of target workers using the context patched by this smaller tp proposer worker.

But if draft and target worker execute _concurrently_ in the future, the code should be redesigned to prevent states being mixed to each other.